openassessment/xblock/studio_mixin.py

```diff
@@ -171,8 +171,17 @@ def update_editor_context(self, data, suffix=''):
 
         # Check that the editor assessment order contains all the assessments.  We are more flexible on example-based.
         if set(DEFAULT_EDITOR_ASSESSMENTS_ORDER) != (set(data['editor_assessments_order']) - {'example-based-assessment'}):
-            logger.exception('editor_assessments_order does not contain all expected assessment types')
-            return {'success': False, 'msg': self._('Error updating XBlock configuration')}
+            # Backwards compatibility: "staff-assessment" may not be present.
+            # If that is the only problem with this data, just add it manually and continue.
+            if set(DEFAULT_EDITOR_ASSESSMENTS_ORDER) == (
+                # Check the given set, minus example-based, plus staff
+                (set(data['editor_assessments_order']) - {'example-based-assessment'}) | {'staff-assessment'}
```

Comment:
Nit: create a variable for set(data['editor_assessments_order']) - {'example-based-assessment'} so the comment can live with it (I was wondering why we don't care about example-based-assessment, but see the comment explaining "we are more flexible" above).